scheduler/filter/constraint_test.go

```diff
@@ -108,3 +112,279 @@ func TestConstrainteFilter(t *testing.T) {
 	assert.NoError(t, err)
 	assert.Len(t, result, 2)
 }
+
+func TestConstraintNotExpr(t *testing.T) {
+	var (
```

Comment:
Do we need a positive/negative test-case using spaces around the operator? `foo == bar`?